frontend/src/components/topicsPage/SelectTreeForm.js

```diff
@@ -3,36 +3,95 @@ import {
   Container,
   Dropdown,
   Box,
+  FormField,
+  Button,
 } from 'wix-style-react';
+import { useDispatch } from 'react-redux';
 import {
-  PERSONAL, MY_TEAM, MY_SUBORDINATES, SPECIFIC_SUBORDINATE, SPECIFIC_TEAM,
+  PERSONAL, MY_TEAM, MY_SUBORDINATES, SINGLE_SUBORDINATE, SINGLE_TEAM,
 } from '../../constants/TreeTypes';
-import {
-  FETCH_SUBORDINATES_SUCCEEDED, FETCH_SUBORDINATES_FAILED,
-} from '../../constants/SubordinatesStatus';
+import { showWarningToast } from '../../state/actions/toast';
+import SelectSubordinateForm from './SelectSubordinateForm';
+import SelectTeamForm from './SelectTeamForm';
 
 const SelectTreeForm = ({ onSelect }) => {
-  const [selectedTree, setSelectedTree] = useState(PERSONAL);
-  const [isDisabled, setIsDisabled] = useState(true);
+  const [tree, setTree] = useState(PERSONAL);
+  const [subordinateId, setSubordinateId] = useState('');
+  const [teamId, setTeamId] = useState('');
+  const [isDisabledSubordinate, setIsDisabledSubordinate] = useState(true);
+  const [isDisabledTeam, setIsDisabledTeam] = useState(true);
+
+  const dispatch = useDispatch();
 
-  const handleDropdownSelect = e => {
-    setSelectedTree(e.id);
+  const handleDropdownSelect = event => {
+    const { value } = event;
+    setTree(value);
+    switch (value) {
+      case SINGLE_SUBORDINATE:
+        setIsDisabledSubordinate(false);
+        setIsDisabledTeam(true);
+        break;
+      case SINGLE_TEAM:
+        setIsDisabledTeam(false);
+        setIsDisabledSubordinate(true);
+        break;
+      default:
+        setIsDisabledSubordinate(true);
+        setIsDisabledTeam(true);
+    }
+  };
+
+  const handleShowBtn = () => {
+    let selectedTree;
+    switch (tree) {
+      case SINGLE_SUBORDINATE:
+        if (subordinateId === '')
```

Comment:
I think this check is not enough. If you choose a subordinate, then you manually change the search field and click on display tree, you'll get tree of the subordinate that you clicked. In other components (create topic modal, edit topic) I changed the logic so that you have to either:
1. Click on a parent topic and do not type anything
2. Leave parent topic empty
In other cases you will get an error toast.
Take a look at `CreateNewTopicForm` and specifically `isSearchAndDropDownMissmatched` (this boolean is true if user typed something after selecting topic)

Reply:
I think this is the most critical issue. Other than that, looks good